tests/Adapters/Microsoft.Bot.Builder.Adapters.Facebook.Tests/FacebookHelperTests.cs

```diff
@@ -0,0 +1,156 @@
+            {
+                Conversation = new ConversationAccount()
+                {
+                    Id = "testId",
+                },
+                Attachments = new List<Attachment> { new Attachment(contentUrl: "http://example.com"), new Attachment(contentUrl: "http://example.com") },
+            };
+
+            Assert.Throws<Exception>(() => { var messageOptions = FacebookHelper.ActivityToFacebook(activity); });
+        }
+
+        [Fact]
+        public void ActivityToFacebookShouldReturnNullWithActivityNull()
+        {
+            Activity activity = null;
+
+            var messageOptions = FacebookHelper.ActivityToFacebook(activity);
+
+            Assert.Null(messageOptions);
+        }
+
+        [Fact]
+        public void ProcessSingleMessageShouldReturnNullWithMessageNull()
+        {
+            FacebookMessage message = null;
+
+            var nullActivity = FacebookHelper.ProcessSingleMessage(message);
+
+            Assert.Null(nullActivity);
+        }
+
+        [Fact]
+        public void ProcessSingleMessageShouldReturnAActivityWithMessageWithData()
+        {
+            var uri = File.ReadAllText(Directory.GetCurrentDirectory() + @"\Files\FacebookMessages.json");
+            var facebookMessage = JsonConvert.DeserializeObject<List<FacebookMessage>>(uri)[0];
+
+            var activity = FacebookHelper.ProcessSingleMessage(facebookMessage);
+            Assert.Equal(activity.Conversation.Id, facebookMessage.Recipient.Id);
+        }
+
+        [Fact]
+        public void ProcessSingleMessageShouldReturnConversationIdWithoutDataOnSender()
+        {
+            var uri = File.ReadAllText(Directory.GetCurrentDirectory() + @"\Files\FacebookMessages.json");
+            var facebookMessage = JsonConvert.DeserializeObject<List<FacebookMessage>>(uri)[1];
+
+            var activity = FacebookHelper.ProcessSingleMessage(facebookMessage);
+            Assert.NotNull(activity.Conversation.Id);
+        }
+
+        [Fact]
+        public void ProcessSingleMessageShouldReturnActivityWithMessage()
+        {
+            var uri = File.ReadAllText(Directory.GetCurrentDirectory() + @"\Files\FacebookMessages.json");
+            var facebookMessage = JsonConvert.DeserializeObject<List<FacebookMessage>>(uri)[2];
+
+            var activity = FacebookHelper.ProcessSingleMessage(facebookMessage);
+            Assert.NotNull(activity.Text);
+            Assert.NotNull(activity.ChannelData);
+        }
+
+        [Fact]
+        public void ProcessSingleMessageShouldReturnActivityWithAttachments()
+        {
+            var uri = File.ReadAllText(Directory.GetCurrentDirectory() + @"\Files\FacebookMessages.json");
+            var facebookMessage = JsonConvert.DeserializeObject<List<FacebookMessage>>(uri)[3];
+
+            var activity = FacebookHelper.ProcessSingleMessage(facebookMessage);
+            Assert.NotNull(activity.Attachments);
+            Assert.Equal(3, activity.Attachments.Count);
```

Comment:
I Think this `3` could go better in a constant. Like this might get confused with the magical number for the index in line 107